### ISSUE TITLE
`type()` is not correctly narrowing type inside `if` block

### ISSUE THREAD

**Issue by michaeloliverx** [microsoft/pylance-release#2263]
## Environment data

-   Language Server version: v2022.1.1
-   OS and version: Windows 10 Enterprise (OS Build: 19043.1348)
-   Python version: 3.9.0

## Expected behaviour

Inside an `if type(value) is date:` block I would expect Pylance to know the type of `value` and provide autocomplete etc.

## Actual behaviour

Using `reveal_type` shows the type as `Any`:

![image](https://user-images.githubusercontent.com/55017335/150126612-031cca82-0e44-4392-a8ce-c8797ea3a7b2.png)

When running `mypy` via the CLI I get:
```
example.py:14: note: Revealed type is "datetime.date"
```

## Logs

```
XXX
```

## Code Snippet / Additional information

```python
from datetime import date, datetime, timezone
from typing import Any

data: dict[str, Any] = {
    "key-1": 1,
    "key-2": datetime.now(timezone.utc),
    "key-3": date(2022, 1, 19),
}


for key, value in data.items():
    if type(value) is date:
        reveal_type(value)
        
```

**Comment by erictraut** [comment]
The `type(x) is y` type narrowing logic currently doesn't perform narrowing if `x` has an `Any` type. In general, `Any` is never narrowed, but there are a few exceptions to this rule. The `isinstance` and `issubclass` narrowing pattern also apply an exception in this case. I think it would be safe to make an exception for the `type(x) is y` narrowing pattern too.

I've created a tracking feature request in the pyright repo: https://github.com/microsoft/pyright/issues/2896.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by michaeloliverx** [comment]
> This will be addressed in the next release.

Thanks for the fast response!

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.1.3, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202213-21-january-2022)
